urllib3/contrib/appengine.py

```diff
@@ -174,17 +179,46 @@ def urlopen(self, method, url, body=None, headers=None,
         http_response = self._urlfetch_response_to_http_response(
             response, retries=retries, **response_kw)
 
-        # Check for redirect response
-        if (http_response.get_redirect_location() and
-                retries.raise_on_redirect and redirect):
-            raise MaxRetryError(self, url, "too many redirects")
+        # Handle redirect?
+        redirect_location = redirect and http_response.get_redirect_location()
+        if redirect_location:
+            # Check for redirect response
+            if (self.urlfetch_retries and retries.raise_on_redirect):
```

Comment:
I believe `raise_on_redirect` works. The `else` block here includes some codes, which is copied from `HTTPConnectionPool`, to raise the error.